storages/storages.go

```diff
@@ -70,6 +70,11 @@ type Backend struct {
 	Maintenance bool
 }
 
+// BackendAdapter holds the necessary backend info after decorating the backend
+type BackendAdapter struct {
+	*Backend
+}
```

Comment:
I don' see the point for having this type, all it looks like no-op extension of Backend